include/rmm/detail/nvtx/ranges.hpp

```diff
@@ -58,4 +58,8 @@ using scoped_range = ::nvtx3::scoped_range_in<librmm_domain>;
  * }
  * ```
  */
+#if defined(RMM_USE_NVTX)
```

Comment:
Some consumers of RMM don't use CMake (for ...reasons?) and just include the headers. If RMM is used by "raw header includes", should NVTX be on or off? This PR changes the default.

Reply:
The idea is that NVTX ranges are close to free when you are not running in a profiler. I thought that was why it was on by default.  The RAPIDS-shared `USE_NVTX` is on by default, and `RMM_USE_NVTX` is set to copy its value by default. So that means the default has not changed IIUC.

However, currently RMM uses an NVTX range on every call to `allocate`/`deallocate`, which is a lot by default, and causes nsys profiles to bloat.  So, perhaps it should not be on every allocation by default. 

Later, we might want to put it on pool growth allocations (for example) by default, since these are likely to be much more expensive.

Reply:
Maybe I can state my question differently: RMM consumers who do not use CMake but instead directly include headers will not have `RMM_USE_NVTX` defined by default, even if the CMake sets it (because they didn't use CMake in the first place).

We have some internal libraries at NVIDIA that I have discovered do not use CMake for including RMM, they just clone a submodule or similar and add the include directory manually.

Reply:
I changed the headers to default to the ranges enabled by default with a disable compile flag to turn off NVTX.  That should preserve the behavior for those not using CMake.  I kept the CMake flag as RMM_USE_NVTX, but I can change it to match the compile flag if that's more desirable.

Reply:
It certainly hurts profiles of our application, which has a deep allocation stack that amplifies the NVTX ranges.  I'm not sure how common that is for RMM applications, and I don't have the full context of why this was enabled by default.  As such, I approached this as preserving the default behavior but allowing an override.

I'll update to disable RMM NVTX ranges by default.